File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
import copy

def solution(triangle):
    max_tri = copy.deepcopy(triangle)
    
    for i, row in enumerate(max_tri[:-1]):
        for j, n in enumerate(row):
            max_tri[i+1][j] = max(max_tri[i+1][j], triangle[i+1][j] + n)
            max_tri[i+1][j+1] = max(max_tri[i+1][j+1], triangle[i+1][j+1] + n)
    return(max(max_tri[-1]))

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        int H = triangle.length;
        
        int[][] new_tri = new int[H][];
        for(int i = 0; i < H; i++) {
            new_tri[i] = new int[triangle[i].length];
        }
        new_tri[0][0] = triangle[0][0];
        
        for(int i = 0; i < H-1; i++) {
            for(int j = 0; j < triangle[i].length; j++) {
                new_tri[i+1][j] = Math.max(new_tri[i+1][j], triangle[i+1][j] + new_tri[i][j]);
                new_tri[i+1][j+1] = Math.max(new_tri[i+1][j+1], triangle[i+1][j+1] + new_tri[i][j]);
            }
        }
        
        return Arrays.stream(new_tri[H-1]).max().getAsInt();
    }
}